File: 692-top-k-frequent-words/692-top-k-frequent-words.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        freq = Counter(words)
        
        heap = []
        d = []
        for i, word in enumerate(words):
            if word not in d:
                d.append(word)
                heap.append((-freq[word], word))
                
        heapq.heapify(heap)
        
        ans = [heapq.heappop(heap)[1] for i in range(k)]
 
        return ans

File: 692-top-k-frequent-words/692-top-k-frequent-words.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        unordered_map<string, int> freq;
        vector<string> ans;
        for (auto w: words)
            freq[w]++;
        priority_queue<pair<int, string>, vector<pair<int, string>>, greater<>> pq;
        for (auto [k, v]: freq)
            pq.emplace(-v, k);
        while (k--) {
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};